lightningd/subd.c

```diff
@@ -191,65 +210,32 @@ static int subd(const char *path, const char *name,
 		goto close_execfail_fail;
 
 	if (childpid == 0) {
-		int fdnum = 3, stdin_is_now = STDIN_FILENO;
 		size_t num_args;
 		char *args[] = { NULL, NULL, NULL, NULL, NULL };
+		int *fds = tal_arr(tmpctx, int, 1);
 
 		close(childmsg[0]);
 		close(execfail[0]);
 
-		// msg = STDIN
-		if (childmsg[1] != STDIN_FILENO) {
-			/* Do we need to move STDIN out the way? */
-			stdin_is_now = dup(STDIN_FILENO);
-			if (!move_fd(childmsg[1], STDIN_FILENO))
-				goto child_errno_fail;
-		}
-
-		/* Dup any extra fds up first. */
+		/* msg = STDIN */
+		fds[0] = childmsg[1];
```

Comment:
How about `STDOUT_FILENO` and `STDERR_FILENO`?  The old version started the passed-in `fd`s at index 3 (xref `fdnum`) aka `STDERR_FILENO + 1`.